airbyte-server/src/main/java/io/airbyte/server/handlers/OAuthHandler.java

```diff
@@ -38,41 +41,53 @@ public OAuthHandler(ConfigRepository configRepository) {
       throws JsonValidationException, ConfigNotFoundException, IOException {
     final OAuthFlowImplementation oAuthFlowImplementation = getSourceOAuthFlowImplementation(sourceDefinitionIdRequestBody.getSourceDefinitionId());
-    return new OAuthConsentRead().consentUrl(oAuthFlowImplementation.getSourceConsentUrl(
+    final ImmutableMap<String, Object> metadata = generateSourceMetadata(sourceDefinitionIdRequestBody.getSourceDefinitionId());
+    final OAuthConsentRead result = new OAuthConsentRead().consentUrl(oAuthFlowImplementation.getSourceConsentUrl(
         sourceDefinitionIdRequestBody.getWorkspaceId(),
         sourceDefinitionIdRequestBody.getSourceDefinitionId(),
         sourceDefinitionIdRequestBody.getRedirectUrl()));
+    TrackingClientSingleton.get().track(sourceDefinitionIdRequestBody.getWorkspaceId(), "OAuth Consent - Backend", metadata);
+    return result;
   }
 
   public OAuthConsentRead getDestinationOAuthConsent(DestinationOauthConsentRequest destinationDefinitionIdRequestBody)
       throws JsonValidationException, ConfigNotFoundException, IOException {
     final OAuthFlowImplementation oAuthFlowImplementation =
         getDestinationOAuthFlowImplementation(destinationDefinitionIdRequestBody.getDestinationDefinitionId());
-    return new OAuthConsentRead().consentUrl(oAuthFlowImplementation.getDestinationConsentUrl(
+    final ImmutableMap<String, Object> metadata = generateDestinationMetadata(destinationDefinitionIdRequestBody.getDestinationDefinitionId());
+    final OAuthConsentRead result = new OAuthConsentRead().consentUrl(oAuthFlowImplementation.getDestinationConsentUrl(
         destinationDefinitionIdRequestBody.getWorkspaceId(),
         destinationDefinitionIdRequestBody.getDestinationDefinitionId(),
         destinationDefinitionIdRequestBody.getRedirectUrl()));
+    TrackingClientSingleton.get().track(destinationDefinitionIdRequestBody.getWorkspaceId(), "OAuth Consent - Backend", metadata);
+    return result;
   }
 
   public Map<String, Object> completeSourceOAuth(CompleteSourceOauthRequest oauthSourceRequestBody)
       throws JsonValidationException, ConfigNotFoundException, IOException {
     final OAuthFlowImplementation oAuthFlowImplementation = getSourceOAuthFlowImplementation(oauthSourceRequestBody.getSourceDefinitionId());
-    return oAuthFlowImplementation.completeSourceOAuth(
+    final ImmutableMap<String, Object> metadata = generateSourceMetadata(oauthSourceRequestBody.getSourceDefinitionId());
+    final Map<String, Object> result = oAuthFlowImplementation.completeSourceOAuth(
         oauthSourceRequestBody.getWorkspaceId(),
         oauthSourceRequestBody.getSourceDefinitionId(),
         oauthSourceRequestBody.getQueryParams(),
         oauthSourceRequestBody.getRedirectUrl());
+    TrackingClientSingleton.get().track(oauthSourceRequestBody.getWorkspaceId(), "OAuth Flow - Backend", metadata);
```

Comment:
are we making tracking a bit too much work for ourselves? why don't we automatically track all the API endpoints that a user interacts with in `ConfigApi` instead of having to add custom code to each handler? maybe not something for this PR but it just stood out to me